[SUPPORT] feature hive sync partition push down fails with glue

hudi 0.14.1

Activating the `hoodie.datasource.hive_sync.filter_pushdown_enabled` leads to errors with aws-sync. similar result in an other thread https://github.com/apache/hudi/issues/8925#issuecomment-1675650288

```
│ Caused by: org.apache.hudi.hive.HoodieHiveSyncException: Partition fields and values should be same length, but got partitionFields: [] with values: [14, 2023-08-22, 08]     │
│     at org.apache.hudi.hive.util.PartitionFilterGenerator.lambda$generatePushDownFilter$5(PartitionFilterGenerator.java:209)                                                  │
│     at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)                                                                                                │
│     at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source)                                                                                    │
│     at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)                                                                                                   │
│     at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)                                                                                            │
│     at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)                                                                                       │
│     at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)                                                                                                   │
│     at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)                                                                                                   │
│     at org.apache.hudi.hive.util.PartitionFilterGenerator.generatePushDownFilter(PartitionFilterGenerator.java:214)                                                           │
│     at org.apache.hudi.hive.HiveSyncTool.getTablePartitions(HiveSyncTool.java:396)                                                                                            │
│     at org.apache.hudi.hive.HiveSyncTool.syncPartitions(HiveSyncTool.java:438)                                                                                                │

```

--- a/hudi-aws/src/main/java/org/apache/hudi/aws/sync/AWSGlueCatalogSyncClient.java
+++ b/hudi-aws/src/main/java/org/apache/hudi/aws/sync/AWSGlueCatalogSyncClient.java
@@ -579,6 +579,14 @@ public class AWSGlueCatalogSyncClient extends HoodieSyncClient {
   }
 
   @Override
+  public List<FieldSchema> getMetastoreFieldSchemas(String tableName) {
+    Map<String, String> schema = getMetastoreSchema(tableName);
+    return schema.entrySet().stream()
+          .map(f -> new FieldSchema(f.getKey(), f.getValue()))
+          .collect(Collectors.toList());
+  }
+
+  @Override
   public boolean tableExists(String tableName) {
     GetTableRequest request = GetTableRequest.builder()
         .databaseName(databaseName)

